src/Database/Type/DecimalType.php

```diff
@@ -158,7 +158,12 @@ public function marshal($value)
             return (float)$value;
         }
 
-        return null;
+        // allow custom decimal format (@see https://github.com/cakephp/cakephp/issues/12800)
```

Comment:
Please remove this comment. You can put the issue link in related test case.

Reply:
done. moved to test cases